adapter/noop2/noop2.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package noop2
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/golang/protobuf/proto"
+
+	adapter "istio.io/mixer/pkg/adapter"
+	adapter_cnfg "istio.io/mixer/pkg/adapter/config"
+	sample_report "istio.io/mixer/pkg/templates/sample/report"
+)
+
+type (
+	noop2Adapter struct{}
+	noop2Builder struct{}
+)
+
+func (noop2Builder) Name() string { return "noop2" }
+func (noop2Builder) Description() string {
+	return "An adapter that does nothing, just echos the calls made from mixer"
+}
+func (noop2Builder) Close() error { return nil }
+
+///////////////// Configuration time Methods ///////////////
+
+func (noop2Builder) DefaultConfig() proto.Message { return &types.Empty{} }
+func (noop2Builder) ValidateConfig(msg proto.Message) error {
+	fmt.Println("ValidateConfig called with input", msg)
+	return nil
+}
+
+func (noop2Builder) ConfigureHandler(cnfg proto.Message) error {
+	fmt.Println("ConfigureHandler in noop Adapter called with", cnfg)
+	return nil
+}
+
+func (noop2Builder) Build() (adapter_cnfg.Handler, error) {
+	fmt.Println("Build in noop Adapter called with")
+	return noop2Adapter{}, nil
+}
+
+// Per template configuration methods
+func (noop2Builder) ConfigureSample(typeParams map[string]*sample_report.Type) error {
+	fmt.Println("ConfigureSample in noop Adapter called with", typeParams)
+	return nil
+}
+
+////////////////// Runtime Methods //////////////////////////
+
+func (noop2Adapter) ReportSample(instances []*sample_report.Instance) error {
+	fmt.Println("ReportSample in noop Adapter called with", instances)
+	return nil
+}
+
+func (noop2Adapter) Close() error { return nil }
+
+// Register registers the no-op adapter as processor for all the templates.
+func Register(r adapter.Registrar2) {
+	r.RegisterSampleProcessor(noop2Builder{})
```

Comment:
This is actually registering a building, so RegisterSampleProcessor is probably not the right name.